[#26488] Moved Droppable code into a separate struct for CanvasRenderingContext2D

willypuzzle opened the pull request:
This is part of incremental implementation of #26488 . The interface involved in  this PR is CanvasRenderingContext2D.

Testing: Test should be just implemented.
Fixes: Partially #26488

jdm commented on components/script/dom/canvasrenderingcontext2d.rs:
```diff
@@ -36,12 +36,35 @@ use crate::dom::path2d::Path2D;
+    fn drop(&mut self) {
+        if let Err(err) = self
+            .canvas_state
+            .get_ipc_renderer()
```
Since the only part of CanvasState that is used in the Drop implementation is the renderer channel, we can simplify this code by only storing a clone of the channel in the DroppableCanvasRenderingContext2D, rather than the entire CanvasState. This also avoids the error you see about CanvsState requiring rooting.

jdm replied:
I actually mean cloning the reference so you are not trying to store a reference.

willypuzzle replied:
@jdm I did it!

willypuzzle committed:
Droppizated CanvasRenderingContext2DMethods

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

Removing clone from get_canvas_id() call

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

sorted derivations

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

added linter exception

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

added linter exception

Signed-off-by: Domenico Rizzo <domenico.rizzo@gmail.com>

jdm reviewed (approved):
Thanks! This looks great.
